pkgs/development/libraries/science/math/magma/generic.nix

```diff
@@ -25,7 +20,9 @@
 , magmaRelease
 , ninja
 , config
-, rocmSupport ? config.rocmSupport
+  # At least one back-end has to be enabled,
+  # and we can't default to CUDA since it's unfree
+, rocmSupport ? !cudaSupport
```

Comment:
I don't think this is a good idea.
We don't necessarily want ``rocmSupport`` to be true for the default derivation.
Yes, we *can* include rocm support for mostly anything because it's not unfree, but that comes with a huge closure cost and someone using magma for non rocm purposes isn't going to want that.
See my reply on samuela's comment for a better solution.

Reply:
There are two possible defaults for `magma`: an evaluation error (mentioning CUDA license) or ROCm. The only non-ROCm purpose magma could serve that I'm aware of is CUDA, and what that is enabled we leave ROCm out.

Note that the CPU-only variant of torch still does not include magma because it's behind the `optionals (cudaSupport || rocmSupport)` guard.

Reply:
Ah, I read the other message first. Well, I don't know, I don't have a strong opinion. I pushed the rebased version that updates the release-notes